tmt/steps/finish/shell.py

```diff
@@ -0,0 +1,63 @@
+import fmf
+import tmt
+import click
+
+
+class FinishShell(tmt.steps.finish.FinishPlugin):
+    """
+    Perform finishing tasks using shell scripts
+
+    Example config:
+
+    finish:
+        how: shell
+        script:
+            - upload-logs.sh
+            - rm -rf /tmp/temporary-files
+
+    Use the 'order' attribute to select in which order finishing tasks
+    should happen if there are multiple configs. Default order is '50'.
```

Comment:
Because finish step is run always, it would be good to document how to avoid uncessary failures here. I.e. if the test fails early, like in prepare phase, finish might again fail as it is trying to cleanup files (or similar) which were created in the executed step. I would document that somewhere and provide an example how to handle that, i.e.

```
finish:
￼        how: shell
￼        script:
￼            - upload-logs.sh || true
￼            - rm -rf /tmp/temporary-files
```

Or we should maybe provide a way how to ignore the failures here ...

Reply:
Looks like a good hint. I've also added a new paragraph to the main `finish` description mentioning that the step is run upon other step failures as well.